testing/testing.go

```diff
@@ -72,6 +72,11 @@ func SetTestHooks(c *gc.C, runner txn.Runner, hooks ...txn.TestHook) Transaction
 	return func() {
 		remaining := <-transactionHooks
 		transactionHooks <- nil
+		// At least one hook has not run. Some common causes:
+		//
+		// - forgetting to prefix the call to Check() with defer
+		// - business logic never succeeding, meaning the transaction operations are never invoked
```

Comment:
Maybe this wording for point 2?
```
- checks in the build function failing, so the slice of txn.Ops never gets run
```
Just to make it more concrete where to look for the problem.

Reply:
Done